plugins/techdocs/src/reader/transformers/handleMetaRedirects.tsx

```diff
@@ -69,15 +68,17 @@ export const handleMetaRedirects = (
         renderReactElement(
           <TechDocsRedirectNotification
             message="This TechDocs page is no longer maintained. Will automatically redirect to the designated replacement."
-            handleButtonClick={() => navigate(redirectURL)}
+            handleButtonClick={() => {
+              window.location.href = redirectURL;
+            }}
             autoHideDuration={redirectAfterMs}
           />,
           container,
         );
         document.body.appendChild(container);
 
         setTimeout(() => {
-          navigate(redirectURL);
+          window.location.href = redirectURL;
```

Comment:
@squid-ney this change was needed to prevent a double redirect happening (can be seen in the video above). Just checking this is OK with you since you added the redirect handling. Thanks!

Reply:
Hi @jroebu14, I think it would be best to avoid a hard page refresh when redirecting so I'd prefer to use the `navigate`. 

This issue is occurring because the `handleMetaRedirects.tsx` function is getting called more than once on the same page with the redirect tag. It should only be called once for that page content. I took a look at why this bug is _only_ occurring when redirecting the `index.md' page and found an edge case. I believe this PR will solve your issue: https://github.com/backstage/backstage/pull/27505 let me know what you think

Reply:
Hey @squid-ney really appreciate you investigating this and providing a fix! PR looks great. Thanks! I'll revert the changes to the redirect handling I've made in this PR.

Reply:
Sounds good, no problem! :)